app/javascript/components/App.js

```diff
@@ -4,35 +4,52 @@ import Home from './pages/Home'
 import AboutUs from './pages/AboutUs'
 import MyAuctions from './pages/MyAuctions'
 import MyBids from './pages/MyBids'
-import CreateAuctions from './pages/CreateAuctions'
-import {
-  BrowserRouter as  Router,
-  Route,
-  Switch
-} from 'react-router-dom'
+import AuctionNew from './pages/AuctionNew'
+import { BrowserRouter as  Router, Route, Switch } from 'react-router-dom'
+import auctions from './mockData'
+import AuctionShow from './pages/AuctionShow'
 
 class App extends Component {
+  constructor(props){
+    super(props)
+    this.state = {
+      auctions: auctions
+    }
+  }
+  createAuction = (auction) => {
+    console.log(auction);
+  }
   render() {
     const {
       logged_in,
       current_user,
       new_user_route,
       sign_in_route,
       sign_out_route,
-      create_auction_route,
+      auction_new_route,
       my_auctions_route,
+      auction_show_route,
       my_bids_route
     } = this.props
 
     return(
       <>
-       
         <Router>
          <Header {...this.props}/>
           <Switch className="nav-bar">
             <Route exact path="/" component={Home} />
-            <Route path="/create_auction_route" component={CreateAuctions} />
-            <Route path="/my_auctions_route" component={MyAuctions} />
+            <Route path="/auction_new_route" component={AuctionNew} />   
+
+            <Route path="/my_auctions_route" render={(props) => <MyAuctions auctions={this.state.auctions} />} />
+            
+            <Route path="/auction_show_route/:id" render={(props) => {
+              let paramId = +props.match.params.id
+              let auction = this.state.auctions.find(auction => auction.id === paramId )
+              return <AuctionShow auction={auction}/>}} />
+
+            
+            <Route path="/auction_new_route" render={(props) => <AuctionNew createAuction={this.createAuction} />} />
```

Comment:
Is this the code that was throwing the error you posted here? https://sdlearn.slack.com/archives/C0325DHSY2V/p1645356049961859

You may need to set the `this` context for the handler. You can do that by using the `bind` method, i.e.:
`<AuctionNew createAuction={this.createAuction.bind(this)} />`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind

Reply:
Since they are using an arrow function you shouldn't need to bind.